My project step 2.py

```diff
@@ -0,0 +1,118 @@
+people = []
+
+def addPerson(name, age, id_number):
+    # Create a dictionary with all the variables
+    person = {'name': name, 'age': age, 'id_number': id_number}
+    people.append(person)
+    print("Added " + name + " with ID " + str(id_number) + " saved successfully.")
+
+def findById():
+    # Getting ID from the user
+    id_number = input("Please enter ID number: ")
+    for person in people:
+        # Check if the selection is inside the variable:
+        if person["id_number"] == id_number:
+            return person
+    return "ID number not found"
+
+def printAgeAverage():
+    if len(people) == 0:
+        print("No data available.")
+        return
+    total_age = 0
+    for person in people:
+        # Sum of ages
+        total_age += person["age"]
+    average_age = total_age / len(people)
+    print("Average age: " + str(average_age))
+
+def printPerson():
+    if len(people) == 0:
+        print("No data available.")
+        return
+    for index, person in enumerate(people):
+        print("Index: " + str(index) + " - Name: " + person['name'] + ", ID: " + person['id_number'])
+
+def printNames():
+    if len(people) == 0:
+        print("No data available.")
+        return
+    for index, person in enumerate(people):
+        print("Index: " + str(index) + " - Name: " + person["name"])
+
+def printID():
+    if len(people) == 0:
+        print("No data available.")
+        return
+    for index, person in enumerate(people):
+        print("Index: " + str(index) + " - ID: " + person["id_number"])
+
+def personByIndex(index):
+    if 0 <= index < len(people):
+        person = people[index]
+        print("Name: " + person["name"] + ", Age: " + str(person["age"]) + ", ID: " + person["id_number"])
+    else:
+        print("Index out of range")
+
+# Using all the last 7 functions in one main function
+def mixAllFunctions():
+    while True:
+        print("1. Save a new entry")
+        print("2. Search person by ID")
+        print("3. Print ages average")
+        print("4. Print all names")
+        print("5. Print all IDs")
+        print("6. Print all entries")
+        print("7. Print entry by index")
+        print("8. Exit")
+        
+        choice = input("Enter your choice (1-8): ")
+        
+        # New entry
+        if choice == "1":
+            name = input("Please enter a name: ")
+            age = int(input("Enter age: "))
+            id_number = input("Enter ID number: ")
+            # This function gets all these variables - name, age, and id_number
+            addPerson(name, age, id_number)
+        
+        # Search by ID
+        elif choice == "2":
+            person = findById()
+            if person == "ID number not found":
```

Comment:
פונקציה תמיד צריכה להחזיר את אותו טיפוס.
במקרה שלך, findById בחלק מהמקרים מחזירה מילון (של אדם) ולפעמים סטרינג "ID number not found". זה לא אותו טיפוס. צריך לתקן